components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1540,6 +1474,10 @@ void RewardsServiceImpl::SetRewardsMainEnabled(bool enabled) {
   SetRewardsMainEnabledPref(enabled);
   bat_ledger_->SetRewardsMainEnabled(enabled);
   TriggerOnRewardsMainEnabled(enabled);
+  greaselion::GreaselionService* greaselion_service =
```

Comment:
I think it would be better to create a separate observer for this using `OnRewardsMainEnabled`. The observers are here so this class doesn't become any more bloated